src/main/java/com/aws/greengrass/clientdevices/auth/iot/usecases/VerifyIotCertificate.java

```diff
@@ -64,6 +64,9 @@ public Boolean apply(String certificatePem) {
             }
         } catch (InvalidCertificateException e) {
             logger.atWarn().kv("certificatePem", certificatePem).log("Unable to process certificate", e);
+            if (cert != null) {
+                certificateRegistry.deleteCertificate(cert);
+            }
```

Comment:
Nit: Not sure if this is ever gonna be used as L61 is going to throw this exception first if the `certificatePem` is invalid. Although L63 can throw this exception for the same reason.

Reply:
Unlikely, but covering all the cases.